pkg/extensions/search/cve/cve.go

```diff
@@ -0,0 +1,234 @@
+package cveinfo
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/anuvu/zot/errors"
+	"github.com/anuvu/zot/pkg/log"
+	integration "github.com/aquasecurity/trivy/integration"
+	config "github.com/aquasecurity/trivy/integration/config"
+	"github.com/aquasecurity/trivy/pkg/report"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+	godigest "github.com/opencontainers/go-digest"
+	ispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// UpdateCVEDb ...
+func UpdateCVEDb(dbDir string, log log.Logger, interval time.Duration, isTest bool) error {
+	config, err := config.NewConfig(dbDir)
+	if err != nil {
+		log.Error().Err(err).Msg("Unable to get config")
+		return err
+	}
+
+	for {
+		log.Info().Msg("Updating the CVE database")
+
+		err = integration.RunTrivyDb(config.TrivyConfig)
+		if err != nil {
+			log.Error().Err(err).Msg("Unable to update DB ")
+			return err
+		}
+
+		if isTest {
+			return nil
+		}
+
+		time.Sleep(interval * time.Hour)
+	}
+}
+
+func NewTrivyConfig(dir string) (*config.Config, error) {
+	return config.NewConfig(dir)
+}
+
+func ScanImage(config *config.Config) (report.Results, error) {
+	return integration.ScanTrivyImage(config.TrivyConfig)
+}
+
+func (cveinfo CveInfo) IsValidImageFormat(imagePath string) (bool, error) {
+	imageDir := getImageDir(imagePath)
+
+	if !dirExists(imageDir) {
+		cveinfo.Log.Error().Msg("Image Directory not exists")
+
+		return false, errors.ErrRepoNotFound
+	}
+
+	buf, err := ioutil.ReadFile(path.Join(imageDir, "index.json"))
+
+	if err != nil {
+		if os.IsNotExist(err) {
+			cveinfo.Log.Error().Err(err).Msg("Index.json does not exist")
+
+			return false, errors.ErrJSONNotFound
+		}
+
+		cveinfo.Log.Error().Err(err).Msg("Unable to open index.json")
+
+		return false, errors.ErrInvalidJSON
+	}
+
+	var index ispec.Index
```

Comment:
Really? the linter is forcing these many whitespaces between lines?

Reply:
Yes